### DISCUSSION TITLE
Do you support streaming generating outputs?

### DISCUSSION THREAD

**Discussion by ltz0120** [vllm-project/vllm#245] [Q&A]

**Comment by WoosukKwon** [comment]
Yes, our FastAPI and OpenAI servers support streaming outputs. Just set up the server with
```bash
python -m vllm.entrypoints.api_server
```
or 
```bash
python -m vllm.entrypoints.openai.api_server
```
and then add `"stream": True` in client request (by default it's false).
See
https://github.com/vllm-project/vllm/blob/665c48963be11b2e5cb7209cd25f884129e5c284/examples/api_client.py#L26

**Comment by zhuohan123** [comment]
In addition, you can see the streaming result from the API server via
```
python vllm/examples/api_client.py --stream
```

**Comment by Bambuuai** [comment]
@WoosukKwon 
Hello, 
I tested 
```bash
python vllm/examples/api_client.py --stream
```
It works. 

But does it support same streaming api as openai?
```bash
curl http://localhost:8000/v1/chat/completions \
    -H "Content-Type: application/json" \
    -d '{
        "model": "facebook/opt-125m",
        "messages": [
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": "Who won the world series in 2020?"}
        ],
       "stream": true
    }'
```

**Comment by nursery42** [comment]
Does streaming output only support return async_generator?

**Comment by Carolmelon** [comment]
你们的文档挺难看懂的感觉，看了半天还是得去看源码😂 可以考虑多加几个demo吗，感谢

**Comment by tempcollab** [comment]
How can I get the usage metrics (number of input tokens and output tokens) in streaming mode? @WoosukKwon
